src/coreclr/inc/allocacheck.h

```diff
@@ -23,7 +23,9 @@
 
 #ifndef AllocaCheck_h
 #define AllocaCheck_h
-#include <malloc.h>			// for alloca itself
+#if !defined(HOST_APPLE)
```

Comment:
Why does it have to be excluded for Apple?

Reply:
Apple doesn't have the `malloc.h` header. It's Windows and POSIX only. This was working before because we had a malloc.h header in our "CRT shim" header set.

Reply:
Ah, I didn't know that. So maybe we should unconditionally include stdlib.h instead, that's the correct header for malloc everywhere.

Reply:
I'll try out removing this header include and seeing which places need it for the POSIX apis it defines.

Reply:
Looks like it was only needed on Windows for `_alloca`.